frontend/src/components/pages/RegistrantExperience/PersonalInfo/CamperCard.tsx

```diff
@@ -0,0 +1,327 @@
+      if (nextBtnRef && nextBtnRef.current) {
+        nextBtnRef.current.removeEventListener("click", updateFormErrorMsgs);
+      }
+    };
+  }, [camper]);
+
+  function AlertDialogExample() {
+    const toast = useToast();
+    const { isOpen, onOpen, onClose } = useDisclosure();
+    const cancelRef = React.useRef(null);
+
+    return (
+      <>
+        <Button
+          color="text.critical.100"
+          variant="outline"
+          onClick={onOpen}
+          colorScheme="blue"
+        >
+          <DeleteIcon />
+          <Text textStyle={{ sm: "xSmallBold", lg: "buttonSemiBold" }} pl={3}>
+            Remove
+          </Text>
+        </Button>
+
+        <AlertDialog
+          isCentered
+          isOpen={isOpen}
+          leastDestructiveRef={cancelRef}
+          onClose={onClose}
+        >
+          <AlertDialogOverlay backgroundColor="overlay.dark.50">
+            <AlertDialogContent>
+              <AlertDialogHeader fontStyle="heading">
+                Remove Camper
+              </AlertDialogHeader>
+
+              <AlertDialogBody>
+                <Text textStyle="bodyRegular">
+                  Are you sure you want to remove this camper from the
+                  registration form? <br />
+                  <br />
+                  <Text as="span" textStyle="bodyBold">
+                    Note: this action is irreversible.{" "}
+                  </Text>
+                </Text>
+              </AlertDialogBody>
+
+              <AlertDialogFooter>
+                <Button ref={cancelRef} onClick={onClose}>
+                  Cancel
+                </Button>
+                <Button
+                  colorScheme="red"
+                  onClick={() => {
+                    setPersonalInfo({
+                      type: PersonalInfoActions.DELETE_CAMPER,
+                      camperId,
+                    });
+                    toast({
+                      description: `${camper.firstName} ${camper.lastName} has been successfully removed`,
+                      status: "success",
+                      duration: 6000,
+                      isClosable: true,
+                      variant: "subtle",
+                    });
+                    onClose();
+                  }}
+                  ml={3}
+                >
+                  Delete
+                </Button>
+              </AlertDialogFooter>
+            </AlertDialogContent>
+          </AlertDialogOverlay>
+        </AlertDialog>
+      </>
+    );
+  }
+
+  return (
+    <Box boxShadow="lg" rounded="xl" borderWidth={1} width="100%">
+      <Box backgroundColor="#FFFFFF" rounded="xl">
+        <Heading textStyle="displayLarge">
+          <Flex py={6} px={{ sm: "5", lg: "20" }} alignItems="center">
+            <Text textStyle={{ sm: "xSmallBold", lg: "displayLarge" }}>
+              Camper #{camperId + 1}
+            </Text>
+            <Spacer />
+            {camperId > 0 ? AlertDialogExample() : null}
```

Comment:
Would it be cleaner if we conditionally render the AlertDialog as part of a top level fragment? 

so the code after the return statement would look like: 
```
return (
<> 
  { camperId > 0 && < alertdialog component here /> }
  < Box component for the actual card here /> 
</>
```

Reply:
I'm not full sure if this would work because the `AlertDialog` should be inside the header of the `Box` component. I think placing it outside would show something like 
<img width="561" alt="Screen Shot 2022-12-19 at 2 53 50 PM" src="https://user-images.githubusercontent.com/117593511/208508798-c9e31dc7-e80b-4c66-9875-aa338c030fc8.png">
<img width="1184" alt="Screen Shot 2022-12-19 at 2 54 02 PM" src="https://user-images.githubusercontent.com/117593511/208508829-3ed2d7c8-f88c-47ab-a1fd-5a6e6e8a9869.png">